autobuild/build.sh

```diff
@@ -51,12 +51,8 @@ if [ -z "$PREFIX" ]; then
 PREFIX=`pwd`/build
 fi
 
-#count of threads for make
-if ( which nproc > /dev/null ); then
-MAKE_THREADS=`nproc`
-else
-MAKE_THREADS=2
-fi
+# detect number of threads for make (this can be overridden by build.conf)
+MAKE_THREADS="$(which nproc >/dev/null && nproc || sysctl -n hw.ncpu || getconf _NPROCESSORS_ONLN || echo 2 2>/dev/null)"
```

Comment:
I do not know operator order/priority for shell; should parenthesis be used surrounding `nproc` stuff?

Reply:
You can read this code like:
```
if (which nproc >/dev/null) then
    echo nproc
else 
    if (sysctl -n hw.ncpu)
        echo sysctl -n hw.ncpu
    else 
        if (getconf _NPROCESSORS_ONLN)
            echo getconf _NPROCESSORS_ONLN
        else
            echo 2
```